source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -421,6 +425,79 @@ void SetupContextForAmazonServiceAccount(string @namespace, string clusterUrl, s
                 var clusterName = variables.Get(SpecialVariables.EksClusterName);
                 log.Info($"Creating kubectl context to {clusterUrl} (namespace {@namespace}) using EKS cluster name {clusterName}");
 
+                log.Verbose("Attempting to authenticate with aws-cli");
+                if (TrySetAws())
+                {
+                    try
+                    {
+                        var region = GetEksClusterRegion(clusterUrl);
+                        if (string.IsNullOrWhiteSpace(region))
+                        {
+                            log.Verbose("The EKS cluster Url specified should contain a valid aws region name");
+                        }
+                        
+                        var awsCliVersion = GetAwsCliVersion();
+
+                        if (new Version(awsCliVersion).CompareTo(new Version("1.16.156")) > 0)
+                        {
+                            var apiVersion = GetEksClusterApiVersion(clusterName, region);
+                            SetKubeConfigAuthenticationToAwsCli(user, clusterName, region, apiVersion);
+                            return;
+                        }
+
+                        log.Verbose($"aws cli version: {awsCliVersion} does not support the \"aws eks get-token\" command. Please update to a version later than 1.16.156");
```

Comment:
A bit of shuffling of things around here can make the code a little more cohesive. Bring the calculation of the Cluster Region down closer to where it's used to make the purpose clearer, eg:

```suggestion                       
                        var awsCliVersion = GetAwsCliVersion();
                        if (new Version(awsCliVersion).CompareTo(new Version("1.16.156")) > 0)
                        {
                            var region = GetEksClusterRegion(clusterUrl);
                            if (!string.IsNullOrWhiteSpace(region))
                            {
                                var apiVersion = GetEksClusterApiVersion(clusterName, region);
                                SetKubeConfigAuthenticationToAwsCli(user, clusterName, region, apiVersion);
                                return;
                            }

                            log.Verbose("The EKS cluster Url specified should contain a valid aws region name");
                        }

                        log.Verbose($"aws cli version: {awsCliVersion} does not support the \"aws eks get-token\" command. Please update to a version later than 1.16.156");
```